62.CAD/fragment_shader.hlsl

```diff
@@ -52,12 +65,29 @@ float4 main(PSInput input) : SV_TARGET
         const float antiAliasingFactor = globals.antiAliasingFactor;
         localAlpha = 1.0f - smoothstep(-antiAliasingFactor, +antiAliasingFactor, distance);
     }
-    /*
-    TODO[Lucas]:
-        Another else case for CurveBox where you simply do what I said in the notes of common.hlsl PSInput
-        and solve two quadratic equations, you could check for it being a "line" for the mid point being nan
-        you will use input.getXXX() to get values needed for this computation
-    */
+    else if (objType == ObjectType::CURVE_BOX) 
+    {
+        float2 positionFullscreen = input.uv;
```

Comment:
notice you're using only `.y` of uv. then just make that a single interpolation input with `get/setMajorAxisValue`